### PR TITLE
Issue 13927 revisited

### PR DESCRIPTION
**Pull request by dsilvam**

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/cluster/business/ClusterAPITest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    @BeforeClass
+    public static void prepare() throws Exception{
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    @DataProvider
+    public static Object[][] testGetReplicasCountDataProvider() {
+        return new Object[][] {{null, true, AUTOWIRE, 0, "false"}, {null, false, STATIC, 1, "false"},
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotcms/cluster/business/ClusterAPITest.java]: I think one case is missing here, when esIndexReplicas=autowire

File: dotCMS/src/main/java/com/dotcms/cluster/business/ClusterAPIImpl.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        this(APILocator.getServerAPI());
+    }
+
+    public ClusterAPIImpl(final ServerAPI serverAPI) {
+        this.serverAPI = serverAPI;
+    }
+
+    @Override
+    public boolean isTransportAutoWire() {
+        return Config.getBooleanProperty("AUTOWIRE_CLUSTER_TRANSPORT", true) && LicenseUtil.getLevel()> 200;
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/cluster/business/ClusterAPIImpl.java]: The hardcoded License Level can be replaced by the enum LicenseLevel.STANDARD

File: dotCMS/src/main/java/com/dotcms/cluster/business/ClusterAPIImpl.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    @Override
+    public boolean isESAutoWire() {
+        return Config.getBooleanProperty("AUTOWIRE_CLUSTER_ES", true)&& LicenseUtil.getLevel()> 200;
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/cluster/business/ClusterAPIImpl.java]: Same here

File: dotCMS/src/main/java/com/dotcms/cluster/business/ClusterAPIImpl.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    @Override
+    public boolean isReplicasSetInConfig() {
+        return UtilMethods.isSet(Config.getStringProperty("ES_INDEX_REPLICAS", null)) && LicenseUtil.getLevel()> 200;
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/cluster/business/ClusterAPIImpl.java]: Same here

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESClient.java [hunk 5]
```diff
@@ -121,11 +132,10 @@ public void setReplicasSettings() {
             //Build the replicas config settings for the indices client
             final Optional<UpdateSettingsRequest> settingsRequest = getReplicasSettings();
 
-            if(settingsRequest.isPresent()) {
-                _nodeInstance.client().admin().indices().updateSettings(
-                        settingsRequest.get()
-                ).actionGet();
-            }
+            settingsRequest.ifPresent(updateSettingsRequest -> _nodeInstance.client().admin().indices().updateSettings(
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESClient.java]: I like it!

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESIndexAPI.java [hunk 6]
```diff
@@ -651,14 +579,10 @@ public synchronized CreateIndexResponse createIndex(final String indexName, Stri
 		}
 		Map map = new ObjectMapper().readValue(settings, LinkedHashMap.class);
 		map.put("number_of_shards", shards);
-		map.put("index.mapping.total_fields.limit", 2000); // TODO ONLY FOR TESTING. REMOVE
+		map.put("index.mapping.total_fields.limit", 5000);
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESIndexAPI.java]: This value should be read from the configuration

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESClient.java [hunk 7]
```diff
@@ -195,34 +199,73 @@ public Builder getESNodeSettings() throws Exception {
...
+
+        final int httpPort = Integer.parseInt(externalSettings
+            .get(ServerPort.ES_HTTP_PORT.getPropertyName()));
+
+        currentServer = Server.builder(currentServer).withEsHttpPort(httpPort).build();
+
+        try {
+            serverAPI.updateServer(currentServer);
+        } catch (DotDataException e) {
+            Logger.error(this,
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESClient.java]: I would include the e.getMessage() to the log.error

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESClient.java [hunk 8]
```diff
@@ -330,18 +342,21 @@ public void restartNode(final Builder externalSettings) {
...
 
-    private String getServerAddress(final Server server) {
-        return  (UtilMethods.isSet(server.getHost()) && !server.getHost().equals("localhost"))
+    @VisibleForTesting
+    protected String getServerAddress(final Server server) {
+        final String ipAddress = (UtilMethods.isSet(server.getHost()) && !server.getHost().equals("localhost"))
             ? server.getHost()
             : server.getIpAddress();
+        final String port = Integer.toString(server.getEsTransportTcpPort());
+        return ipAddress + ":" + port;
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESClient.java]: : can be replaced by StringPool.COLON

File: dotCMS/src/test/java/com/dotcms/content/elasticsearch/util/ESClientTest.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+public class ESClientTest {
+
+    @Test
+    public void testgetServerAddress_ValidIpAndValidPort_ShouldGetAddressSemiColonPort() {
+        final String ipAddress = "192.168.1.186";
+        final int port = 9300;
+        final Server server = Server.builder().withIpAddress(ipAddress).withEsTransportTcpPort(port).build();
+        ESClient esClient = new ESClient();
+        final String fullAddress = esClient.getServerAddress(server);
+        final String expectedAddress = ipAddress + ":" + port;
```
Comment (nollymar) [inline dotCMS/src/test/java/com/dotcms/content/elasticsearch/util/ESClientTest.java]: : can be replaced by StringPool.COLON

File: dotCMS/src/test/java/com/dotcms/content/elasticsearch/util/ESClientTest.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class ESClientTest {
+
+    @Test
+    public void testgetServerAddress_ValidIpAndValidPort_ShouldGetAddressSemiColonPort() {
+        final String ipAddress = "192.168.1.186";
```
[the remark this answers is not in the document]
Comment (dsilvam) [reply dotCMS/src/test/java/com/dotcms/content/elasticsearch/util/ESClientTest.java]: this is ok, I just need whatever address to test

**Review by nollymar — APPROVED** [review]

**Pull request merged**